localstack/testing/pytest/fixtures.py

```diff
@@ -1040,14 +1041,18 @@ def _destroy_stack():
 
     for entry in state:
         entry_stack_id = entry.get("stack_id")
-        entry_change_set_id = entry.get("change_set_id")
         try:
-            entry_change_set_id and cleanup_changesets([entry_change_set_id])
-            entry_stack_id and cleanup_stacks([entry_stack_id])
+            if entry_stack_id:
+                aws_client.cloudformation.delete_stack(StackName=entry_stack_id)
+                aws_client.cloudformation.get_waiter(WAITER_STACK_DELETE_COMPLETE).wait(
+                    StackName=entry_stack_id,
+                    WaiterConfig={
+                        "Delay": 2,
+                        "MaxAttempts": 120,
+                    },
+                )
         except Exception as e:
-            LOG.debug(
-                f"Failed cleaning up change set {entry_change_set_id=} and stack {entry_stack_id=}: {e}"
-            )
+            LOG.debug(f"Failed cleaning up stack {entry_stack_id=}: {e}")
```

Comment:
How often does this line show up in the logs? It would be interesting to see how often this cleanup fails, since that may leave resources around

Reply:
Actually I think we are a bit too lenient with the delete success status when deleting stacks, I haven't found a single case where this is logged in CI :thinking: 

There are a few cases where deletion seems to fail though, e.g.
```
2023-08-30T07:42:43.410 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 2. Retrying in next cycle.
2023-08-30T07:42:43.419 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 2. Retrying in next cycle.
2023-08-30T07:42:43.428 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 3. Retrying in next cycle.
2023-08-30T07:42:43.438 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 3. Retrying in next cycle.
2023-08-30T07:42:43.447 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 4. Retrying in next cycle.
2023-08-30T07:42:43.457 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 4. Retrying in next cycle.
2023-08-30T07:42:43.467 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 5. Retrying in next cycle.
2023-08-30T07:42:43.476 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 5. Retrying in next cycle.
2023-08-30T07:42:43.487 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 6. Retrying in next cycle.
2023-08-30T07:42:43.497 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 6. Retrying in next cycle.
2023-08-30T07:42:43.506 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 7. Retrying in next cycle.
2023-08-30T07:42:43.516 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 7. Retrying in next cycle.
2023-08-30T07:42:43.526 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 8. Retrying in next cycle.
2023-08-30T07:42:43.536 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 8. Retrying in next cycle.
2023-08-30T07:42:43.546 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id FilePublishingRoleDefaultPolicy in iteration cycle 9. Retrying in next cycle.
2023-08-30T07:42:43.555 WARNING --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Failed delete of resource with id ImagePublishingRoleDefaultPolicy in iteration cycle 9. Retrying in next cycle.
2023-08-30T07:42:43.565 ERROR --- [  asgi_gw_13] localstack.services.cloudformation.engine.template_deployer : Last cycle failed to delete resource with id FilePublishingRoleDefaultPolicy. Final exception: An error occurred (NoSuchEntity) when calling the DeletePolicy operation: Policy stack-cdk--54UC7WVIBG855 was not found.
```

We should investigate this a bit further. I think the whole delete stack process is currently a bit broken